pkg/prometheus/promcfg.go

```diff
@@ -590,6 +591,12 @@ func (cg *configGenerator) generatePodMonitorConfig(
 		})
 	}
 
+	// Relabel job_name into a meta label
+	relabelings = append(relabelings, yaml.MapSlice{
```

Comment:
Another option would be to initialize `relabelings` like this:

```
relabelings = []yaml.MapSlice{
  {Key: "source_labels", Value: []string{"job"}},
  {Key: "target_label", Value: "__tmp_prometheus_job_name"}
}
```

which would avoid the need for the `jobName` variable.